File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let indices = new Map();
    for (let i = 0; i < nums.length; i++) {
        let com = target - nums[i];
        if (indices.has(com)) return [i, indices.get(com)];
        indices.set(nums[i], i);
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # loop through all the num, keep track of previous numbers & idcs
        # if current number can be added to a prev num to == target, then we found match
        num_to_idx = {}
        for i in range(len(nums)):
            comp = target - nums[i]
            if comp in num_to_idx:
                return [num_to_idx[comp], i]
            num_to_idx[nums[i]] = i
        

